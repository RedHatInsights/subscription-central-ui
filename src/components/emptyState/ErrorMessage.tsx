import React, { FC } from 'react';
import {
  Button,
  EmptyState,
  EmptyStateBody,
  EmptyStateIcon,
  EmptyStateVariant,
  Title
} from '@patternfly/react-core';
import ExclamationCircleIcon from '@patternfly/react-icons/dist/js/icons/exclamation-circle-icon';

const ErrorMessage: FC = () => {
  return (
    <EmptyState variant={EmptyStateVariant.small}>
      <EmptyStateIcon icon={ExclamationCircleIcon} color="var(--pf-global--danger-color--100)" />
      <Title headingLevel="h3">Something went wrong</Title>
      <EmptyStateBody>
        <h4>
          Try refreshing the page. If the problem persists, contact your organization administrator
          or visit our{' '}
          <Button
            variant="link"
            component="a"
            href="https://status.redhat.com/"
            target="_blank"
            isInline
          >
            status page
          </Button>{' '}
          for known outages.
        </h4>
      </EmptyStateBody>
    </EmptyState>
  );
};

export default ErrorMessage;
