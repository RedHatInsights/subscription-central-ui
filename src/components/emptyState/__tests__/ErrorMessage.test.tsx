import React from 'react';
import { render } from '@testing-library/react';
import ErrorMessage from '../ErrorMessage';

describe('Generic Error Message', () => {
  it('renders correctly', () => {
    const { container } = render(<ErrorMessage />);
    expect(container).toMatchSnapshot();
  });
});
