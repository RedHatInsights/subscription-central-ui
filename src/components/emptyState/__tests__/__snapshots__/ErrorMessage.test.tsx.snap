// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generic Error Message renders correctly 1`] = `
<div>
  <div
    class="pf-c-empty-state pf-m-sm"
  >
    <div
      class="pf-c-empty-state__content"
    >
      <svg
        aria-hidden="true"
        class="pf-c-empty-state__icon"
        fill="var(--pf-global--danger-color--100)"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 512 512"
        width="1em"
      >
        <path
          d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
        />
      </svg>
      <h3
        class="pf-c-title pf-m-lg"
      >
        Something went wrong
      </h3>
      <div
        class="pf-c-empty-state__body"
      >
        <h4>
          Try refreshing the page. If the problem persists, contact your organization administrator or visit our
           
          <a
            aria-disabled="false"
            class="pf-c-button pf-m-link pf-m-inline"
            data-ouia-component-id="OUIA-Generated-Button-link-1"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            href="https://status.redhat.com/"
            target="_blank"
          >
            status page
          </a>
           
          for known outages.
        </h4>
      </div>
    </div>
  </div>
</div>
`;
